src/main/java/org/spongepowered/api/world/explosion/Explosion.java

```diff
@@ -44,41 +42,20 @@
     World getWorld();
 
     /**
-     * Sets the world the explosion will occur in.
+     * Gets the source {@link Explosive} of this explosion.
      *
-     * @param world The world
-     */
-    void setWorld(World world);
-
-    /**
-     * Gets the source {@link Explosive} of the explosion.
-     *
-     * @return The source explosive or null if there is no source
+     * @return The source explosive, if present
```

Comment:
If this is a usability, you should note that you're making `Explosion` immutable.